scripts/ci/check_compliance.py

```diff
@@ -804,6 +827,31 @@ def check_disallowed_defconfigs(self, kconf):
                                "--perl-regexp", regex_socs, "--", ":soc",
                                cwd=ZEPHYR_BASE)
 
+        manifest = Manifest.from_file()
+        for project in manifest.get_projects([]):
+            if not manifest.is_active(project):
+                continue
+
+            if not project.is_cloned():
+                continue
+
+            module_path = PurePath(project.abspath)
+            module_yml = module_path.joinpath('zephyr/module.yml')
+
+            if not Path(module_yml).is_file():
+                module_yml = module_path.joinpath('zephyr/module.yaml')
+
+            if Path(module_yml).is_file():
+                with Path(module_yml).open('r', encoding='utf-8') as f:
+                    meta = yaml.load(f.read(), Loader=SafeLoader)
+
+                    if 'build' in meta and 'settings' in meta['build']:
+                        grep_stdout_boards += self.module_disallowed_check(module_path, 'board',
```

Comment:
how about having this split by lines after the initial assignment in line 823, then here you can do an extend and down below you can just iterate on the list, feels wired to implement `module_disallowed_check` to return a single string rather than some structured data just to make it compatible with the git output